composite-view/README.md

```diff
@@ -0,0 +1,325 @@
+            <h3>
+                Generic Local News
+            </h3>
+            <ul style="list-style-type: none">
+                <li>
+                    Mayoral elections coming up in 2 weeks
+                </li>
+                <li>
+                    New parking meter rates downtown coming tomorrow
+                </li>
+                <li>
+                    Park renovations to finish by the next year
+                </li>
+                <li>
+                    Annual marathon sign ups available online
+                </li>
+            </ul>
+        </div>
+    </body>
+</html>
+```
+The results are as such:
+
+1) The user has put their name as `Tammy` in the request parameters and no preferences: 
+![alt text](etc/images/noparam.PNG)
+2) The user has put their name as `Johnny` in the request parameters and has a preference for world, business, and science news:
+![alt text](etc/images/threeparams.PNG)
+
+The different subviews such as `worldNews.jsp`, `businessNews.jsp`, etc. are included conditionally
+based on the request parameters.
+
+**How To Use**
+
+To try this example, make sure you have Tomcat 10+ installed.
+Set up your IDE to build a WAR file from the module and deploy that file to the server
+
+IntelliJ:
+
+Under `Run` and `edit configurations` Make sure Tomcat server is one of the run configurations. 
+Go to the deployment tab, and make sure there is one artifact being built called `composite-view:war exploded`. 
+If not present, add one.
+
+Ensure that the artifact is being built from the content of the `web` directory and the compilation results of the module.
+Point the output of the artifact to a convenient place. Run the configuration and view the landing page, 
+follow instructions on that page to continue.
+
+## Class diagram
+
+![alt text](./etc/composite_view.png)
+
+The class diagram here displays the Javabean which is the view manager.
+The views are JSP's held inside the web directory.
+
+## Applicability
+
+This pattern is applicable to most websites that require content to be displayed dynamically/conditionally.
+If there are components that need to be re-used for multiple views, or if the project requires reusing a template, 
+or if it needs to include content depending on certain conditions, then this pattern is a good choice.
+
+## Known uses
+
+Most modern websites use composite views in some shape or form, as they have templates for views and small atomic components
+that are included in the page dynamically. Most modern Javascript libraries, like React, support this design pattern 
+with components.
+
+## Consequences
+**Pros**
+* Easy to re-use components
+* Change layout/content without affecting the other
+* Reduce code duplication
+* Code is more maintainable and modular
+
+**Cons**
+* Overhead cost at runtime
+* Slower response compared to directly embedding elements
+* Increases potential for display errors 
+
+## Related patterns
+* [Composite (GoF)](../composite)
```

Comment:
I suggest linking to the website https://java-design-patterns.com/patterns/composite/

Reply:
apologies, I was busy for the holidays, I have added these changes in my latest commit.